### ISSUE TITLE
default code rendering is not monospace

### ISSUE THREAD

**Issue by randy3k** [facelessuser/sublime-markdown-popups#24]
When I run the this,
```python
mdpopups.show_popup(view, mdpopups.syntax_highlight(view, "office = 'office'", language="python"))
```
the following pop up appears,
<img width="140" alt="screen shot 2016-12-07 at 9 22 03 pm" src="https://cloud.githubusercontent.com/assets/1690993/20995177/44c5ad8a-bcc3-11e6-9722-2e50b4bae9c4.png">

Note that the code is not displayed as monospace (the `fi` term).

**Comment by facelessuser** [comment]
Yes, I am aware that it doesn't **force** monospace by default, but this is because it doesn't set the font at all.  Popups and phantoms should use the font that your editor is currently set to.  I always use a monospace as my default font; therefore my font in the popups is monospaced.  Sublime does this by default.

![screenshot 2016-12-07 19 45 35](https://cloud.githubusercontent.com/assets/1055125/20995681/dc221096-bcb5-11e6-8c5c-811f41e51484.png)

Now, you might wonder why I don't just set the font to monospaced for code elements.  Well, when I was experimenting https://github.com/facelessuser/mdpopup_test (which is based off of https://github.com/aziz/sublime-mini-ui), I found that using the font property would cause unexpected crashes.  I don't know if this is because I was using multiple fonts in the document or not, but removing the font properties brought back stability.  Now I haven't played with it on recent ST versions, but I am going to start folding in some of the things from the experiment in the next version.

If the font property is more stable now, I have no problem forcing monospace, but if not, people are going to have to deal with non-monospace when their system is set to non-monospace (or I guess if a tmTheme forces something different).  I'd rather have stability over pretty fonts that cause instability.  Again, I'm not sure if the issue was introduced by using lot's of different fonts in the popup or if just using one alternate font for the whole popup would still cause the issue.  I will have to play with this again.

**Comment by randy3k** [comment]
I see. Thank you for such a detailed explanations! Switching to default font does give me the expected result.

I thought that my font is monospace, but it seems it is not truly monospace. All other characters are just fine.

<img width="344" alt="screen shot 2016-12-07 at 10 19 32 pm" src="https://cloud.githubusercontent.com/assets/1690993/20996388/3ffa3c32-bccb-11e6-8f7f-44e73ef73f3d.png">

except that it makes `fi` shorter, 

<img width="50" alt="screen shot 2016-12-07 at 10 20 14 pm" src="https://cloud.githubusercontent.com/assets/1690993/20996409/5b6f1384-bccb-11e6-8bdf-2977553cb73a.png">

**Comment by randy3k** [comment]
The following solves my issue.
```python
code = mdpopups.syntax_highlight(view, "office", language="python")
mdpopups.show_popup(view, "<div style='font-family:monospace'>{}</div>".format(code))
```

 I guess it is reasonable to force monospace font for code blocks.

**Comment by facelessuser** [comment]
> I guess it is reasonable to force monospace font for code blocks.

Only if it is stable.  Like I said, I would play with it and see how stable it is in the long term.  It worked, but then I started getting random crashes.  Hopefully this is fixed in recent builds.

> It is not a good solution because it changes font...

You need to use the css parameter and pass in CSS to target the code blocks:

```python
code = mdpopups.syntax_highlight(view, "office", language="python")
css = '''
.myplugin pre, .myplugin code {
  font-family:monospace;
}
'''
mdpopups.show_popup(
    view,
    code,
    wrapper_class="myplugin",
    css=css
)
```

I haven't tested the above, but something close to that should work.

**Comment by randy3k** [comment]
Brilliant, it works perfectly!

**Comment by facelessuser** [comment]
If it turns out to be stable, I will do this by default.

**Comment by facelessuser** [comment]
> EDIT, it still changes font. Anyway, I guess it is not an issue for mkpops per se. Close it now.

Did you mean Sublime overrides your code blocks?  Or Sublime changes all the other text except your code block?  I'd have to play around and double check the CSS, but it should be possible to target your code blocks unless some other CSS is being injected from somewhere (like from a tmTheme file).  Anyways, I will play with this before the next release.  If I can get it working and it is stable, then I will force a monospace font on code blocks.

**Comment by randy3k** [comment]
I don't know if I understand what you meant, but it is what happening.

I set my sublime font face to `Courier` and then run this in console
```python
code = mdpopups.syntax_highlight(view, "office", language="python")
css = '''.myplugin pre, .myplugin code { font-family:monospace; } '''
mdpopups.show_popup(view, code, wrapper_class="myplugin", css=css)
```

![screen shot 2016-12-08 at 1 25 52 pm](https://cloud.githubusercontent.com/assets/1690993/21022259/036ae666-bd4a-11e6-9dd9-62e034c6efd7.png)
Note that the font is not `Courier` as in the background.

**Comment by facelessuser** [comment]
I must have misunderstood your edit statement.  That is what I would expect.

**Comment by randy3k** [comment]
It is possible to force it to use the rendering font if it is monospace already?

**Comment by facelessuser** [comment]
Yeah, it is technically possible.  You could keep a list of known monospaced fonts and check the current set font against that and conditionally include the CSS to change the font if the current font is not in that list (you could even pass in template variables into the mdpopups function to do the conditional logic there).  You could also use something like the python fontforge module to detect if the font is monospaced.  That is all a lot of work though for very little gain.

This is how mdpopups is planning to approach it (maybe in the very next version that is to release within the week):

I plan to provide a default css rule that defines a couple of valid monospaced font options that it can attempt to go through.  Maybe:

```
font-family: "sf mono", "Liberation Mono", Menlo, Courier, monospace;
```

And then let the user override it in there mdpopups user css.

In short, I plan to provide something reasonable, and then let the user override it with something specific if they so choose (in this case the current font they are using).

No matter what you provide, there is going to be someone who wants to tweak the look.  This is why I encourage all plugin developers to provide a plugin wrapper class and/or very unique class names so the user can target them.  Mdpopups aims to provide something consistent and useful out of the box in the hopes to limit how much CSS a developer would need to do (but obviously some plugins will have very specific needs that can't be provided).
